src/IO/ProcessRunner.php

```diff
@@ -13,4 +13,6 @@
      * @return Process
      */
     public function run(Process $process, $mustSucceed = true);
+
+    public function setUserInteraction(UserInteraction $userInteraction);
```

Comment:
I know this cannot be set in the container because it's not available at that point but using a setter for this leaves it in a potential invalid state since the process runner is expected to be set in some of the implementations of the run method.

This could just be ignored, a check for it added to the run implementations or something done to only construct this with the UserInteraction object. The latter would probably mean introducing a factory (or factories if per process runner) that would be injected and then require the UserInteraction to be passed to it to return a valid ProcessRunner.